### DISCUSSION TITLE
Use a single PR for all dependency upgrades, except for package rules

### DISCUSSION THREAD

**Discussion by tomkerkhove** [renovatebot/renovate#10379] [Request Help] [answered]
We currently have the following package rules:
```json
{
  "packageRules": [
    {
      "packagePatterns": ["^k8s.io/","^sigs.k8s.io/"],
      "groupName": "Kubernetes Dependencies"
    },
    {
      "packagePatterns": ["^actions/"],
      "groupName": "GitHub Actions"
    }
  ],
}
```

However, we want to have a single PR for the rest of the dependencies.

After going through the docs I've noticed that you can use a [single PR for all dependencies](https://docs.renovatebot.com/faq/#use-a-single-branchpr-for-all-dependency-upgrades) but was wondering if I can use a similar approach, but still have separate PRs for the ones above.

Is that possible?

Relates to https://github.com/kedacore/keda/pull/1871

**Comment by HonkingGoose** [comment]
You might want to follow issue #4404, as that will give you the ability to "roll up multiple PRs into one big PR", which seems to me to be a part of what you want the bot to do with your config?

I'll let the Renovate config wizards help with your config question.

**Comment by viceice** [comment, marked as answer]
You can simply add the group all rule before your more precise rules, so last matching rule wins.

**Reply by tomkerkhove** [reply]
Oh that's great, thank you! Something like this then?

```json
{
  "packageRules": [
    {
      "packagePatterns": ["^k8s.io/","^sigs.k8s.io/"],
      "groupName": "Kubernetes Dependencies"
    },
    {
      "packagePatterns": ["^actions/"],
      "groupName": "GitHub Actions"
    },
    {
      "packagePatterns": ["^"],
      "groupName": "All"
    }
  ],
}
```

**Reply by viceice** [reply]
The all rule must come first or it will override the previous.

**Reply by viceice** [reply]
You should use * instead of ^

**Reply by tomkerkhove** [reply]
Great, thanks a ton!

**Reply by tomkerkhove** [reply]
*looks for "Mark as answered"*

**Reply by HonkingGoose** [reply]
> looks for "Mark as answered"

That button is only enabled on "Questions & Answers" type discussions unfortunately. 😉
